src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import './Carousel.scss';
+import { State } from '../../types/State';
+import { Direction } from '../Direction';
+
+export const Carousel: React.FC<State> = ({
+  images,
+  itemWidth,
+  frameSize,
+  step,
+  animationDuration,
+  infinity,
+}) => {
+  const [indexStart, setIndexImage] = useState(0);
+
+  useEffect(() => {
+    if (images.length > frameSize) {
+      setIndexImage(prevIndexStart => prevIndexStart - 1);
+    }
+  }, [frameSize]);
+
+  const handleMoveRight = () => {
+    const isEnoughImages = indexStart + step >= images.length - frameSize;
+
+    if (isEnoughImages && !infinity) {
+      setIndexImage(images.length - frameSize);
+    } else if (isEnoughImages && infinity) {
+      setIndexImage(0);
+    } else {
+      setIndexImage(indexStart + step);
+    }
```

Comment:
you can use early return for better readabilty
```suggestion
    if (isEnoughImages && !infinity) {
      setIndexImage(images.length - frameSize);
      return;
    } 
    
    if (isEnoughImages && infinity) {
      setIndexImage(0);
      return;
    }
    
    setIndexImage(indexStart + step);
```